server/controllers/item.js

```diff
@@ -0,0 +1,123 @@
+const Item = require('../models/item');
+const List = require('../models/list');
+const User = require('../models/user');
+
+
+// adding item
+exports.addItem = (req, res ) => {
+    const { name, url, list } = req.body;
+    const item = new Item({
+        name, url, user: req.user._id
+    })
+
+    try {
+        List.findById(list._id)
+        .populate('items')
+        .populate('user')
+        .exec(async function(err, foundList) {
+            if(err) return res.status(400).json({
+                error: 'Invalid data!'
+            });
+
+            foundList.items.push(item); // add to items
+            // update item
+            item.list = foundList;
+
+
+            // save item 
+            await item.save();
+            await foundList.save();
+            // update user model
+            // await User.updateOne({ 
+            //     _id: req.user._id}, 
+            //     {$push: { items: item }
+            // }, function(){});
+            }
+        );
+    
+        res.status(200).json(item);
+    } catch(e) {
+        return res.status(500).send(e.message);
+    }
+}
+
+// getAllItems
+exports.getAllItems = async (req, res) => {
+    try {
+        await req.user.populate('items').execPopulate();
+
+        res.status(200).send(req.user.items);
+    } catch(e) {
+        res.status(500).send(e);
+    }
+}
+
+
+// get single item
+exports.getSingleItem = async (req, res) => {
+    const _id = req.params.id;
+
+    try {
+        const item = await Item.findOne({ _id, user: req.user._id });
+
+        if(!item) res.status(401).json({
+            error: 'Item is not found!'
+        });
+
+        // return item
+        res.status(200).json(item);
+    } catch(e) {
+        res.status(500).send(e);
+    }
+}
+
+
+// Update item
+exports.updateItem = async (req, res) => {
+    // check for validation updates
+    const updates = Object.keys(req.body);
+    const allowedUpdateFields = ["name", "url"];
```

Comment:
I would define this somewhere else outside of the function, since it's a constant and it doesn't have to be redefined each time the function is called